airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -340,15 +356,34 @@ def _run(self, kube_client, resource_version, worker_uuid, kube_config):
         watcher = watch.Watch()
 
         kwargs = {'label_selector': 'airflow-worker={}'.format(worker_uuid)}
+
         if resource_version:
             kwargs['resource_version'] = resource_version
+
         if kube_config.kube_client_request_args:
             for key, value in kube_config.kube_client_request_args.items():
                 kwargs[key] = value
 
-        last_resource_version = None
-        for event in watcher.stream(kube_client.list_namespaced_pod, self.namespace,
-                                    **kwargs):
+        if resource_version:
+            last_resource_version = resource_version
+        else:
+            last_resource_version = None
+
+        event_generator = watcher.stream(kube_client.list_namespaced_pod,
+                                         self.namespace,
+                                         **kwargs)
+        while True:
+            try:
+                event = next(event_generator)
+            except StopIteration:
+                break
+            except ReadTimeoutError:
+                self.log.info("Timed out waiting for an event.")
+                break
+            except HTTPError:
+                self.log.info("Terminating connection to kube-api.")
+                break
```

Comment:
for HTTPError, consult https://github.com/astronomer/issues/issues/387

Reply:
Should we merge if this causes a busy loop for a few seconds, logging a lot before getting terminated?

Reply:
@schnie let me know what you think on this one, otherwise I think this is good to go. I'm going to start doing Astro commits like this [ASTRO-###] to reference the astronomer issue number. That will help us during the upstreaming process, I think.